components/dropdown/dropdown-content-mixin.js

```diff
@@ -0,0 +1,556 @@
+import { clearDismissible, setDismissible } from '../../helpers/dismissible.js';
+import { findComposedAncestor, isComposedAncestor } from '../../helpers/dom.js';
+import { getComposedActiveElement, getFirstFocusableDescendant, getPreviousFocusableAncestor } from '../../helpers/focus.js';
+import { html } from 'lit-element/lit-element.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+export const DropdownContentMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			minWidth: {
+				type: Number,
+				reflect: true,
+				attribute: 'min-width'
+			},
+			maxWidth: {
+				type: Number,
+				reflect: true,
+				attribute: 'max-width'
+			},
+			noAutoClose: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-auto-close'
+			},
+			noAutoFit: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-auto-fit'
+			},
+			noAutoFocus: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-auto-focus'
+			},
+			noPadding: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-padding'
+			},
+			noPointer: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-pointer'
+			},
+			align: {
+				type: String,
+				reflect: true
+			},
+			boundary: {
+				type: Object,
+			},
+			opened: {
+				type: Boolean,
+				reflect: true
+			},
+			/**
+			 * Private.
+			 */
+			openedAbove: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'opened-above'
+			},
+			/**
+			 * Whether to render the content immediately. By default, the content rendering
+			 * is deferred.
+			 */
+			renderContent: {
+				type: Boolean,
+				attribute: 'render-content'
+			},
+			verticalOffset: {
+				type: String,
+				attribute: 'vertical-offset'
+			}
+		};
+	}
+
+	constructor() {
+		super();
+
+		this.__opened = false;
+		this.__content = null;
+		this.__isRTL = false;
+		this.__previousFocusableAncestor = null;
+		this.__applyFocus = true;
+		this.__dismissibleId = null;
+
+		this.__onResize = this.__onResize.bind(this);
+		this.__onAutoCloseFocus = this.__onAutoCloseFocus.bind(this);
+		this.__onAutoCloseClick = this.__onAutoCloseClick.bind(this);
+		this.__toggleScrollStyles = this.__toggleScrollStyles.bind(this);
+	}
+
+	get opened() {
+		return this.__opened;
+	}
+
+	set opened(val) {
+		const oldVal = this.__opened;
+		this.__opened = val;
+		this.requestUpdate('opened', oldVal);
+		this.__openedChanged(val);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.__content = this.__getContentContainer();
+		this.__content.addEventListener('scroll', this.__toggleScrollStyles);

```

Comment:
Looks like this `scroll` handler could be wired up down in `_renderContentContainer` in the template... `@scroll...`.